spec/unit/rust-crypto/rust-crypto.spec.ts

```diff
@@ -1395,6 +1398,261 @@ describe("RustCrypto", () => {
+            const outgoingRequestsManager = new OutgoingRequestsManager(logger, olmMachine, outgoingRequestProcessor);
+            rustCrypto["outgoingRequestsManager"] = outgoingRequestsManager;
+
+            async function createSecretStorageKey() {
+                return {
+                    keyInfo: {} as AddSecretStorageKeyOpts,
+                    privateKey: new Uint8Array(32),
+                };
+            }
+
+            // create initial secret storage
+            await rustCrypto.bootstrapCrossSigning({ setupNewCrossSigning: true });
+            await rustCrypto.bootstrapSecretStorage({
+                createSecretStorageKey,
+                setupNewSecretStorage: true,
+                setupNewKeyBackup: false,
+            });
+
+            // there isn't any dehydrated device yet
+            fetchMock.config.overwriteRoutes = true;
+            fetchMock.get("path:/_matrix/client/unstable/org.matrix.msc3814.v1/dehydrated_device", {
+                status: 404,
+                body: {
+                    errcode: "M_NOT_FOUND",
+                    error: "Not found",
+                },
+            });
+            expect(await rustCrypto.rehydrateDeviceIfAvailable()).toBe(false);
+
+            // create a dehydrated device
+            let dehydratedDeviceBody: any;
+            fetchMock.put("path:/_matrix/client/unstable/org.matrix.msc3814.v1/dehydrated_device", (_, opts) => {
+                dehydratedDeviceBody = JSON.parse(opts.body as string);
+                return {};
+            });
+            await rustCrypto.createAndUploadDehydratedDevice();
+
+            // rehydrate the device that we just created
+            fetchMock.get("path:/_matrix/client/unstable/org.matrix.msc3814.v1/dehydrated_device", {
+                device_id: dehydratedDeviceBody.device_id,
+                device_data: dehydratedDeviceBody.device_data,
+            });
+            const eventsResponse = jest.fn((url, opts) => {
+                // rehydrating should make two calls to the /events endpoint.
+                // The first time will return a single event, and the second
+                // time will return no events (which will signal to the
+                // rehydration function that it can stop)
+                const body = JSON.parse(opts.body as string);
+                const nextBatch = body.next_batch ?? "0";
+                const events =
+                    nextBatch === "0" ? [{ sender: testData.TEST_USER_ID, type: "m.dummy", content: {} }] : [];
+                return {
+                    events,
+                    next_batch: nextBatch + "1",
+                };
+            });
+            fetchMock.post(
+                `path:/_matrix/client/unstable/org.matrix.msc3814.v1/dehydrated_device/${encodeURIComponent(dehydratedDeviceBody.device_id)}/events`,
+                eventsResponse,
+            );
+
```

Comment:
I see no mention of `POST /dehydrated_device/claim` from [MSC2697](https://github.com/matrix-org/matrix-spec-proposals/pull/2697) - is that not included?

Reply:
Ah, wrong MSC! This is the right link: [MSC3814](https://github.com/matrix-org/matrix-spec-proposals/pull/3814)